### PR TITLE
2dec sendgrid integration

### PR DESCRIPTION
**Pull request by himankbhalla**
@championswimmer  Please review and merge. Will have to add sendgrid key to env and create  transactional templates for welcome and verifying emails in sendgrid. And add those template id's. Change the sender's email in config.

### CODE REVIEW

**Comment by championswimmer** [comment]
this looks good, 
on a different topic, on second thought, we should use HBS templates, as that will make us less dependent on Sendgrid, and easily let us switch mailing provider later on. 

We should have HBS templates, and compile HTML from them and send those HTMLs as emails. This way, in future we can use SMTP native, or other nodemailer drivers if we want to switch from Sendgrid

**Comment by championswimmer** [comment]
@witty123 
Would like your views on this - 

Sendgrid allows creating templates on their GUI, but that way our entire project becomes dependent on Sendgrid. I think oneauth should be an independent system, regardless of what mailing service we use. 
I believe we should compose our own HTML emails using HBS templates.

**Comment by witty123** [comment]
@championswimmer agreed that oneauth should be independent of any mailer service provider but given that we're on a tight schedule to release things I feel that currently we should leverage what sendgrid is providing to us. We won't go in too deep and create thousands of templates here, will keep it to a level where we can easily move out of it in another 2-3 months.

**Comment by championswimmer** [comment]
There's a shortcut.

Build template on Sendgrid GUI.  Download it as HTML (they allow it I think)
If not,  then let's go ahead with current impl

Sent from Outlook Mobile<https://aka.ms/blhgte>

________________________________

**Comment by witty123** [comment]
Yeah we can download as html anytime we want to move out.

File: config.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ config = {
   "AUTH_TOKEN_SIZE": 64,
   "BCRYPT_SALT_ROUNDS": 8,
   "DEBUG": false,
-  "NEWRELIC_LOG_LEVEL": "info"
+  "NEWRELIC_LOG_LEVEL": "info",
+  "EMAIL_SENDER_ADDR": "himank.bhalla@codingblocks.com"
```
Comment (championswimmer) [inline config.js]: should be info@codingblocks.com

**Comment by championswimmer** [comment]
@himankbhalla make the 2 noted changes, and i'll merge this

**Pull request merged**